akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -455,13 +456,29 @@ final case class `Content-Location`(uri: Uri) extends jm.headers.ContentLocation
  * two there is slight but important difference regarding how parameter values are formatted. In RFC6266 parameters values are without quotes and
  * in RFC2616 they are quoted. Since common practice among http servers is to understand quoted values, we use older document
  * as reference here.
+ * Extended (i.e. encoded) parameter values are not quoted, see ext-parameter - https://tools.ietf.org/html/rfc5987#section-3.2.1
  */
 object `Content-Disposition` extends ModeledCompanion[`Content-Disposition`]
 final case class `Content-Disposition`(dispositionType: ContentDispositionType, params: Map[String, String] = Map.empty)
   extends jm.headers.ContentDisposition with RequestResponseHeader {
   def renderValue[R <: Rendering](r: R): r.type = {
+    import Rendering.{ contentDispositionFilenameSafeChars => safeChars }
     r ~~ dispositionType
-    params foreach { case (k, v) => r ~~ "; " ~~ k ~~ '=' ~~#! v }
+    val renderExtFilename =
+      params.get("filename").map(safeChars.indexOfFirstMismatch(_) >= 0) getOrElse false
+    val withExtParams =
+      if (renderExtFilename && !params.contains("filename*"))
```

Comment:
I think we should document that render logic somewhere. So, it is:

 * if `filename` exists and contains unsafe chars => also render `filename*`
 * if explicit `filename*` exists => don't generate `filename*` automatically

Reply:
> if `filename` exists and contains unsafe chars => also render `filename*`

IIRC there's some contention about whether to render both `filename*` and `filename`. Is there any data about what is best (basically, how browsers interpret those)? Does order matter?

Reply:
https://tools.ietf.org/html/rfc6266#appendix-D suggests they should both be rendered and `filename` should come first - but I'm not sure how outdated that advice might be.

Reply:
> if filename exists and contains unsafe chars => also render filename*

This is because of the decision to hide `filename*` as implementation detail and put unicode under `filename` key in model. Then spec requires to render `filename*`

> if explicit filename* exists => don't generate filename* automatically

This is to allow control over ascii `filename` without resorting to `RawHeader`. For example, provide `"telephone"` instead of default `"t?l?phone"` (derived automatically from `"téléphone"`). It would also be expected to correctly render parameters mentioned in spec.

Reply:
> This is because of the decision to hide `filename*` as implementation detail and put unicode under `filename` key in model. Then spec requires to render `filename*`

Makes sense.

> > if explicit filename* exists => don't generate filename* automatically
> 
> This is to allow control over ascii `filename` without resorting to `RawHeader`. For example, to provide `"telephone"` instead of default `"t?l?phone"` (derived automatically from `"téléphone"`). It would also be expected behaviour to correctly render both parameters mentioned in spec, if provided explicitly.

I agree, that's nice when needed.